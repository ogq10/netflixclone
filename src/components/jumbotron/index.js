import React from 'react';
import {
  Container,
  Item,
  Inner,
  Title,
  Left,
  Right,
  SubTitle,
  Image,
} from './styles/jumbotron';

export default function Jumbotron({
  children,
  direction = 'row',
  ...restProps
}) {
  return (
    <Item {...restProps}>
      <Inner direction={direction}>{children}</Inner>
    </Item>
  );
}

Jumbotron.Container = function JumbotronContainer({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
};

Jumbotron.Left = function JumbotronContainer({ children, ...restProps }) {
  return <Left {...restProps}>{children}</Left>;
};

Jumbotron.Title = function JumbotronContainer({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

Jumbotron.SubTitle = function JumbotronContainer({ children, ...restProps }) {
  return <SubTitle {...restProps}>{children}</SubTitle>;
};

Jumbotron.Right = function JumbotronContainer({ children, ...restProps }) {
  return <Right {...restProps}>{children}</Right>;
};

Jumbotron.Image = function JumbotronContainer({ ...restProps }) {
  return <Image {...restProps} />;
};
