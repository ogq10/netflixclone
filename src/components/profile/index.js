import React from "react";
import { Container, Title, Item, List, Name, Picture } from "./styles/profile";

export default function Profile({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

Profile.Title = function ProfileTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

Profile.User = function ProfileUser({ children, ...restProps }) {
  return <Item {...restProps}>{children}</Item>;
};

Profile.List = function ProfileList({ children, ...restProps }) {
  return <List {...restProps}>{children}</List>;
};

Profile.Name = function ProfileName({ children, ...restProps }) {
  return <Name {...restProps}>{children}</Name>;
};

Profile.Picture = function ProfilePicture({ src, ...restProps }) {
  return (
    <Picture
      {...restProps}
      src={src ? `/images/users/${src}.png` : `/images/misc/loading.png`}
    />
  );
};
