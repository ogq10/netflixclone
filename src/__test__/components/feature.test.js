import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import { Feature } from "../../components";

describe("<Feature/>", () => {
  it("renders the <Feature/> with populated data", () => {
    const { container } = render(
      <Feature>
        <Feature.Title>Unlimited films, TV programmes and more.</Feature.Title>
        <Feature.SubTitle>
          Watch anywhere. Cancel at any time.
        </Feature.SubTitle>
      </Feature>
    );

    expect(screen.getByText("Unlimited films, TV programmes and more.")).toBeTruthy();
    expect(screen.getByText("Watch anywhere. Cancel at any time.")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
});
});
