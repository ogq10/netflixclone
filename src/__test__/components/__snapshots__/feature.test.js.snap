// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Feature/> renders the <Feature/> with populated data 1`] = `
<div
  class="feature__Container-sc-1dljsz-0 hUtytl"
>
  <h1
    class="feature__Title-sc-1dljsz-1 eseGen"
  >
    Unlimited films, TV programmes and more.
  </h1>
  <h2
    class="feature__SubTitle-sc-1dljsz-2 kpbCbg"
  >
     
    Watch anywhere. Cancel at any time.
  </h2>
</div>
`;
